src/decode.rs

```diff
@@ -1442,14 +1472,7 @@ unsafe fn decode_b(
     }
 
     let ts = &mut *t.ts;
-    let f = &mut *t.f;
-    let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
-    let mut b_mem = Default::default();
-    let b = if t.frame_thread.pass != 0 {
-        &mut f.frame_thread.b[(t.by as isize * f.b4_stride + t.bx as isize) as usize]
-    } else {
-        &mut b_mem
-    };
+    let bd_fn = f.bd_fn.clone();
```

Comment:
```suggestion
    let bd_fn = &f.bd_fn;
```

Doesn't appear to cause any borrowck issues.  And if it did, it could be derived from `f.bitdepth_max` and `const` `Rav1dFrameContext_bd_fn`s, which can be created as `const`s.

Reply:
Actually, since `&f.bd_fn` works, `f.bd_fn` can be used everywhere in this `fn`, so no change is needed there at all.

Reply:
New version, `bd_fn` not needed:

```suggestion
    let seq_hdr = &***f.seq_hdr.as_ref().unwrap();
    let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
```